packages/ui/src/components/molecules/DropDownListItem.tsx

```diff
@@ -0,0 +1,194 @@
+  showCheckBox?: boolean;
+  id?: string;
+  onClick?: () => void;
+  selectedItem?: string | null;
+  shouldChangeColor?: boolean;
+  checked?: boolean;
+  onCheckedChange?: (checked: boolean) => void;
+}
+
+export interface DropDownListItemHorizontalBoxProps {
+  isChecked: boolean;
+}
+
+export const DropDownListItemHorizontalBox = styled.div<DropDownListItemHorizontalBoxProps>`
+  display: flex;
+  padding: 12px 24px;
+  align-items: center;
+  gap: 16px;
+  align-self: stretch;
+  border-bottom: 1px solid #2c2824;
+  background-color: ${({ isChecked }) => (isChecked ? '#272320' : '#46403c')};
+  &:hover {
+    background-color: #272320;
+  }
+`;
+
+export const DropDownListItemIconContainer = styled.div`
+  display: flex;
+  align-items: center;
+`;
+
+export const DropDownListItemStretchedTypography = styled(Typography)<{
+  shouldStretch: boolean;
+  changeColor?: boolean;
+}>`
+  flex: ${({ shouldStretch }) => (shouldStretch ? '1' : 'unset')};
+  color: 'white';
+`;
+
+export const DropDownListItemRightContent = styled.div`
+  margin-left: auto;
+  display: flex;
+  align-items: center;
+  gap: 16px;
+`;
+
+export const DropDownListItem: FC<DropDownListItemProps> = ({
+  leftImageSrc,
+  rightIconSrc,
+  displayRadioButton,
+  radioButtonValue,
+  rightText,
+  selectedItem,
+  text,
+  tag,
+  textVariant = 'fineprint',
+  rightTextVariant = 'fineprint',
+  rightTextColor = 'gold',
+  showCheckBox = false,
+  checked = false,
+  id,
+  onClick,
+  restrictedItem = false,
+  shouldChangeColor = false,
+  onCheckedChange,
+}): ReactElement => {
+  const [isChecked, setChecked] = useState(false);
+  const [isHovered, setIsHovered] = useState(false);
+
+  const handleMouseEnter = () => {
+    setIsHovered(true);
+  };
+
+  const handleMouseLeave = () => {
+    setIsHovered(false);
+  };
+
+  const handleCheckBoxChange = () => {
+    const newChecked = !isChecked;
+    setChecked(newChecked);
+
+    if (onCheckedChange) {
+      onCheckedChange(newChecked);
+    }
+  };
+
+  const handleRadioButtonChange = () => {
+    // state code to be added here
+  };
+
+  return (
+    <DropDownListItemHorizontalBox
+      onClick={onClick}
+      isChecked={checked}
+      onMouseEnter={handleMouseEnter}
+      onMouseLeave={handleMouseLeave}
+    >
+      {!restrictedItem && displayRadioButton && (
+        <RadioButton
+          checked={selectedItem === id}
+          value={radioButtonValue}
+          onChange={handleRadioButtonChange}
+        />
+      )}
+      {leftImageSrc && (
+        <DropDownListItemIconContainer>
+          <Image
+            src={leftImageSrc}
+            alt="Left Icon"
+            width="20px"
+            height="16px"
+          />
+        </DropDownListItemIconContainer>
+      )}
+      <DropDownListItemStretchedTypography
+        shouldStretch={!tag}
+        variant={textVariant}
+        changeColor={shouldChangeColor || isHovered}
+      >
+        {text}
+      </DropDownListItemStretchedTypography>
```

Comment:
1. Why can't we use css `:hover` to change the color?
2. `changeColor` prop does not seem to change anything here.

Reply:
updated the code, it was missed in the push